plugins/last-fm.py

```diff
@@ -0,0 +1,237 @@
+    url += f'&artist={track_artist}&api_key={api_key}'
+    response = requests.get(url)
+    if response.status_code == 200:
+        try:
+            tags_json = response.json()['toptags']['tag'][0:5]
+        except IndexError:
+            tags_json = response.json()['toptags']['tag']
+
+        if len(tags_json) < 1:
+            return 'none'
+
+        tags = []
+        for tag in tags_json:
+            tags.append(tag['name'])
+        tags = ', '.join(tags)
+        return tags
+
+
+def _get_when_played(track):
+    """returns the time difference between right now and the time the
+    track was played in text format"""
+
+    """i decided to not include this in the output since the API
+     is either buggy, or is reacting too slowly to changes. i.e a track
+     i had just scrobbled shows up as 'played one hour ago' """
+    try:
+        listened = datetime.strptime(track['date']['#text'], '%d %b %Y, %H:%M')
+        curtime = datetime.fromtimestamp(time.time())
+        curtime = datetime.strptime(str(curtime), '%Y-%m-%d %H:%M:%S.%f')
+        diff = str((curtime - listened))
+        listened = diff.split('.')[0]
+        if listened[0:2] == '-1':
+            listened = listened[8:-6] + ' seconds'
+        elif listened[0] == '0':
+            if listened[2:4] == '01':
+                listened = listened[3:-3] + ' minute'
+            elif listened[2:3] == '0':
+                if listened[2:4] == '00':
+                    listened = '1 minute'
+                else:
+                    listened = listened[3:-3] + ' minutes'
+            else:
+                listened = listened[2:-3] + ' minutes'
+        else:
+            if listened[-8:-6] == ' 1' or listened[0:2] == '1:':
+                listened = listened[:-6] + ' hour'
+            else:
+                listened = listened[:-6] + ' hours'
+        return listened
+    except KeyError:
+        pass
+
+
+def _pull_track_plays(api_key, track_title, track_artist, lastfm_nick):
+    url = api_url+f'&method=track.getinfo&username={lastfm_nick}'
+    url += f'&track={track_title}&artist={track_artist}'
+    url += f'&api_key={api_key}'
+    response = requests.get(url)
+    if response.status_code == 200:
+        try:
+            playcount_json = response.json()['track']['userplaycount']
+            return str(playcount_json)
+        except:
+            return 'unknown'
+
+
+def _output_nowplaying(client, data, lastfm_nick,
+                       title, artist,album, plays, tags):
+    out = f'{lastfm_nick} is listening to "{title}" by {artist} from '
+    if album == 'an unknown album':
+        out += 'an unknown album'
+    else:
+        out += f'the album {album}'
+
+    out += f', Play count: {plays}, Tags: {tags}.'
+
+    asyncio.create_task(
+        client.message(
+            data.target, out))
+
+    return
+
+
+@hook.hook('command', ['lastfminit'], admin=True)
+async def lfminit(client, data):
```

Comment:
You don't need a whole table just for the lastfm, you can just add a column to the users table with the lastfm name. `        db.add_column(conn, 'users', 'lastfm')`, you could add this to [admin.py](https://github.com/FrozenPigs/Taigabot3/blob/master/plugins/admin.py#L50) where I got some db stuff for the channels, or you could just add it at the top of the command or something like that.